### ISSUE TITLE
Fix VizDoom resolution and inputs on the server

### ISSUE THREAD

**Issue by Serenamidori** [schrum2/MM-NEATv2#233]
When running VizDoom on the server, the input substrates and coordinates (for HyperNEAT and NEAT respectively) are correct, but the actual inputs are not. My current theory is that the resolution of the game and screen is not the one we assigned, which is backed but the fact that the screen is visibly larger (I originally thought this was just Window 10 trying to make it more visible, but that doesn't make sense now) and this also explains why it's been "learning" so slowly on the server than before.

I'll either make a parameter to choose the resolution we'd like, or figure out a way to permanently have it at 150x200 for testing always. I really don't think we'll ever need a higher or lower resolution going forward on the server.

**Comment by schrum2** [comment]
The lowest resolution should definitely be our default.

Does this resolution issue only occur on the server/Windows 10? Very
curious.

On Sun, Jun 26, 2016 at 10:23 AM, Gab notifications@github.com wrote:

> When running VizDoom on the server, the input substrates and coordinates
> (for HyperNEAT and NEAT respectively) are correct, but the actual inputs
> are not. My current theory is that the resolution of the game and screen is
> not the one we assigned, which is backed but the fact that the screen is
> visibly larger (I originally thought this was just Window 10 trying to make
> it more visible, but that doesn't make sense now) and this also explains
> why it's been "learning" so slowly on the server than before.
> 
> I'll either make a parameter to choose the resolution we'd like, or figure
> out a way to permanently have it at 150x200 for testing always. I really
> don't think we'll ever need a higher or lower resolution going forward on
> the server.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/schrum2/MM-NEATv2/issues/233, or mute the thread
> https://github.com/notifications/unsubscribe/AIAhtLd-srAOsOOyGysOrQq3iWhsIhqRks5qPpmFgaJpZM4I-kxn
> .

## 

Jacob Schrum, Ph.D.
Assistant Professor of Computer Science
Department of Mathematics and Computer Science
Southwestern University
1001 E. University Ave.
Georgetown, TX 78626

Office: Mood-Bridwell Hall 141
Mail: SU Box 7341
Off-Campus Phone: (512) 863-1712
On-Campus Extension: x1712
E-mail: schrum2@southwestern.edu
WWW: http://southwestern.edu/~schrum2

**Comment by Serenamidori** [comment]
Not sure yet, it hasn't ever happened on the computer I work on now, but I'm wondering if that's always been an issue on the server computer. I'll go ahead and work on it today too and update on what I figure out.

**Comment by Serenamidori** [comment]
Update so far- the method where we set the restricted screen resolution is printing out that the resolution is 200x150 like it should, but the window is clearly bigger on the server. The input window is still the correct smaller size, so I'm guessing we're getting a chunk of the bigger screen and that's why the agent can't see anything- it's just looking at the walls. The only thing is, I don't know why the resolution is changing- It's either Windows 10, the fact that we're only moving parts of the repository, or something is changing after we do and before the testing starts.

I'll need to ask someone to test if they can run any VizDoom tasks on their computers and get the correct screen resolution and readable inputs, because it may be an issue of being on a specific system vs. windows 10 itself.

For now, I'll check if there's any way that the code is resetting the resolution after we do.

**Comment by schrum2** [comment]
Completely clearing out the server files and re-uploading them is probably
a good idea. In particular, make sure the server has the latest version of
the vizdoom directory.

**Comment by Serenamidori** [comment]
Unfortunately, that doesn't change anything. I reloaded the files to the server, and made sure everything was the same (excluding src and bin folders because we never moved those over) and I get the same result. I took screen shots as well to explain what I think is going on.

I added two print statements to the VizDoom Task constructor- both after the resolution is set. The checks make sure our selected resolution is what's chosen to be the resolution of the game. However- even though the server side run of VizDoomDefendCenter prints out the correct resolution (200 by 150), it displays a bigger window showing VizDoom when watch is on and on PostWatch.

I took out the row portion of the batch file code to allow the entire screen to be smudged instead of just a portion, and this is what shows up:

![wrong](https://cloud.githubusercontent.com/assets/16706141/16363748/fabce118-3b99-11e6-87d1-10ee7720edbc.png)
![wrong2](https://cloud.githubusercontent.com/assets/16706141/16363749/fabd8ef6-3b99-11e6-90af-52def3b900b3.png)

Obviously- the resolution we set, check for, and get correct in terms of input size and substrates is not the resolution the VizDoom is actually evolving with. My first thought is that it's being set somewhere else in the code, but I'm actually at a loss of where. Not only that- but why only on the server? When I can get someone to confirm the inputs work on their computer tomorrow, I'll be able to make more progress on this myself, but this is all I have for now.
